Source/Icebreaker/Helpers/IcebreakerBotDataProvider.cs

```diff
@@ -67,6 +69,57 @@ public async Task UpdateTeamInstallStatusAsync(TeamInstallInfo team, bool instal
             }
         }
 
+        /// <summary>
+        /// Get a list of past pairings
+        /// </summary>
+        /// <returns>List of past pairings.</returns>
+        public async Task<IList<PairInfo>> GetPairHistoryAsync()
+        {
+            await this.EnsureInitializedAsync();
+
+            var pairHistory = new List<PairInfo>();
+
+            try
+            {
+                using (var lookupQuery = this.documentClient
+                    .CreateDocumentQuery<PairInfo>(this.pairsCollection.SelfLink, new FeedOptions { EnableCrossPartitionQuery = true })
+                    .AsDocumentQuery())
+                {
+                    while (lookupQuery.HasMoreResults)
+                    {
+                        var response = await lookupQuery.ExecuteNextAsync<PairInfo>();
+                        pairHistory.AddRange(response);
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                this.telemetryClient.TrackException(ex.InnerException);
+            }
+
+            return pairHistory;
+        }
+
+        /// <summary>
+        /// Record a pairing that was made
+        /// </summary>
+        /// <param name="pair">A pairing.</param>
+        /// <param name="lastIteration">Value that indicates the iteration cycle when the pairing happened.</param>
+        /// <returns>Tracking task</returns>
+        public async Task AddPairAsync(Tuple<string, string> pair, int lastIteration)
+        {
+            await this.EnsureInitializedAsync();
+
+            var pairInfo = new PairInfo
+            {
+                User1Id = pair.Item1,
+                User2Id = pair.Item2,
+                Iteration = lastIteration,
+            };
+
+            var response = await this.documentClient.UpsertDocumentAsync(this.pairsCollection.SelfLink, pairInfo);
```

Comment:
Since `lastIteration` is always increasing and there isn't the possibility of collisions (or updates) consider using `CreateDocumentAsync`.